spynnaker/pyNN/models/neuron/connection_holder.py

```diff
@@ -48,11 +48,17 @@ class ConnectionHolder(object):
 
         # A callback to call with the data when finished
         "__notify",
+
+        # The app edge where the connections are read from
+        "__app_edge",
+
+        # The number of machine edges that have been filled in
+        "__n_machine_edges"
     )
 
     def __init__(
             self, data_items_to_return, as_list, n_pre_atoms, n_post_atoms,
-            connections=None, fixed_values=None, notify=None):
+            connections=None, fixed_values=None, notify=None, app_edge=None):
```

Comment:
rethinking this. it can be avoided and needs fixing due to a problem with reset. 

but current thinking is that you have a separate algorithm which runs after data loaded token completed, whos job is to call finish ON all holders that havent alread been called finish. thus you need a has_finished flag, and a reset fiunish flag, which reset will need to reset. 

this fixes both a bug in master and gets rid off this app edge merlarky.

Reply:
Fixed by having an additional algorithm